plutarch-ledger-api/src/Plutarch/LedgerApi/V1/Contexts.hs

```diff
@@ -0,0 +1,59 @@
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Plutarch.LedgerApi.V1.Contexts (
+  PScriptPurpose (..),
+) where
+
+import Plutarch.DataRepr (
+  DerivePConstantViaData (DerivePConstantViaData),
+ )
+import Plutarch.LedgerApi.V1.Credential qualified as Credential
+import Plutarch.LedgerApi.V1.DCert qualified as DCert
+import Plutarch.LedgerApi.V1.Tx qualified as Tx
+import Plutarch.LedgerApi.Value qualified as Value
+import Plutarch.Lift (
+  PConstantDecl,
+  PUnsafeLiftDecl (PLifted),
+ )
+import Plutarch.Prelude
+import PlutusLedgerApi.V1 qualified as Plutus
+
+-- | @since 3.1.1
+data PScriptPurpose (s :: S)
```

Comment:
Any particular reason why we moved this to its own module? 

I think having own module for this is better convention, but just wondering if there's other reason

Reply:
Ah nvm, V2 has to import this. Makes senes